src/SFA.DAS.AdminService.Web.Tests/Controllers/Gateway/OrganisationChecks/IcoNumberTests.cs

```diff
@@ -0,0 +1,140 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Newtonsoft.Json;
+using NUnit.Framework;
+using SFA.DAS.AdminService.Web.Controllers.Roatp.Apply;
+using SFA.DAS.AdminService.Web.Infrastructure;
+using SFA.DAS.AdminService.Web.Services.Gateway;
+using SFA.DAS.AdminService.Web.Validators.Roatp;
+using SFA.DAS.AdminService.Web.ViewModels.Roatp.Gateway;
+using SFA.DAS.AssessorService.Api.Types.Models.Validation;
+using SFA.DAS.AssessorService.ApplyTypes.Roatp;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using SFA.DAS.AdminService.Web.Infrastructure.RoatpClients;
+
+namespace SFA.DAS.AdminService.Web.Tests.Controllers.Gateway.OrganisationChecks
+{
+    [TestFixture]
+    public class IcoNumberTests
+    {
+        private RoatpGatewayOrganisationChecksController _controller;
+        private Mock<IRoatpApplicationApiClient> _applyApiClient;
+        private Mock<IHttpContextAccessor> _contextAccessor;
+        private Mock<IRoatpGatewayPageViewModelValidator> _gatewayValidator;
+        private Mock<IGatewayOrganisationChecksOrchestrator> _orchestrator;
+        private Mock<ILogger<RoatpGatewayOrganisationChecksController>> _logger;
+
+        private string username = "john smith";
+        private string givenName = "john";
+        private string surname = "smith";
+
+        [SetUp]
+        public void Setup()
+        {
+            _applyApiClient = new Mock<IRoatpApplicationApiClient>();
+            _contextAccessor = new Mock<IHttpContextAccessor>();
+            _gatewayValidator = new Mock<IRoatpGatewayPageViewModelValidator>();
+            _logger = new Mock<ILogger<RoatpGatewayOrganisationChecksController>>();
+            _orchestrator = new Mock<IGatewayOrganisationChecksOrchestrator>();
+
+            var user = new ClaimsPrincipal(new ClaimsIdentity(new Claim[]
```

Comment:
We do this setup in a lot of unit test classes now, potential for this to be moved into a base class?

Reply:
Definitely.  I'll need to do it on my next page, happy if I do it then?  I'll go back and change any existing tests as well.

Reply:
Yep, works for me.